xarray/core/groupby.py

```diff
@@ -380,6 +383,19 @@ def __init__(
         # cached attributes
         self._groups = None
 
+        if len(group_indices) == 0:
+            if bins is not None:
+                raise ValueError(
+                    "None of the data falls within bins with edges %r" % bins
+                )
+            else:
+                raise ValueError(
+                    "Failed to group data. Are you grouping by a variable that is all NaN?"
+                )
```

Comment:
Is it possible to hit this second error? I thought line 324 would have caught this?

Reply:
This catches grouping by a variable that is all NaN. That line catches `bins` being all NaN.